### DISCUSSION TITLE
how to read my own .csv files of train/evaluation/test

### DISCUSSION THREAD

**Discussion by dyoflut** [RUCAIBox/RecBole#1120] [Q&A]
Once I generated train/evaluation/test interactions and saved them as .csv files, then how can I load them into RecBole for reuse?

**Comment by Wicknight** [comment]
Hello! The data formats we support include six atomic files. For details, see [doc](https://recbole.io/docs/user_guide/data/atomic_files.html). So you need to convert your raw data into Atomic Files format before data loading, see [doc](https://recbole.io/docs/user_guide/usage/running_new_dataset.html). And after the conversion, you can use your own dataset by setting parameter 'benchmark_filename' in config, see [doc](url[)](https://recbole.io/docs/user_guide/config/data_settings.html) and issue #1114

**Reply by dyoflut** [reply]
Thanks a lot~I'm confident it'll work out. But beforehand I also tried another seemingly roundabout strategy: 

1) train/evaluation/test .csv files → DataFrame → Dataset,  one by one through pandas, Interaction() and dataset.copy(). 

2) copy and modify the original 'data_preparation' function as 

built_datasets = [train_data_Dataset, valid_data_Dataset, test_data_Dataset]
train_sampler, valid_sampler, test_sampler = create_samplers(config, dataset, built_datasets)
train_data = get_dataloader(config, 'train')(config, train_data_Dataset, train_sampler, shuffle=True)
valid_data = get_dataloader(config, 'evaluation')(config, valid_data_Dataset, valid_sampler, shuffle=False)
test_data = get_dataloader(config, 'evaluation')(config, test_data_Dataset, test_sampler, shuffle=False)
dataloaders = (train_data, valid_data, test_data)

3) Finally save such dataloaders as .pth files.

I think the above process works. But it meanwhile generates .pth files slightly different from the original .csv files about negative item_ids, because these ids have been recreated through '*_sampler' of the above process.

Could u plz help me check the process? Thanks a lot~

**Reply by Wicknight** [reply]
Hello! I think there should be no problem with your handling. Could you please provide more information about the difference about negative item_ids?
